### PR TITLE
[9.x] Enforce stricter assertions

### PR DESCRIPTION
**Pull request by lucasmichot**
This PR enforces stricter string assertions with `assertSame`.

### CODE REVIEW

File: tests/Auth/AuthenticatableTest.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ public function testItReturnsStringAsRememberTokenWhenItWasSetToTrue()
     {
         $user = new User;
         $user->setRememberToken(true);
-        $this->assertSame('1', $user->getRememberToken());
+        $this->assertSame('1',   $user->getRememberToken());
```
Comment (driesvints) [inline tests/Auth/AuthenticatableTest.php:21]: Can you undo this?

File: tests/Database/DatabaseEloquentBelongsToManySyncReturnValueTypeTest.php [hunk 2]
```diff
@@ -90,7 +90,7 @@ public function testSyncReturnValueType()
         });
 
         $user->articles->each(function (BelongsToManySyncTestTestArticle $article) {
-            $this->assertEquals('0', $article->pivot->visible);
+            $this->assertSame('0', (string)  $article->pivot->visible);
```
Comment (driesvints) [inline tests/Database/DatabaseEloquentBelongsToManySyncReturnValueTypeTest.php:93]: All of your `(string)` casting have two spaces after them. Can you reduce them to only one?

Comment (lucasmichot) [reply tests/Database/DatabaseEloquentBelongsToManySyncReturnValueTypeTest.php]: done

**Comment by ankurk91** [comment]
@lucasmichot
You are on fire :fire:

**Comment by lucasmichot** [comment]
> @lucasmichot You are on fire 🔥

😆

**Pull request merged**